### ISSUE TITLE
Random Pits To Nowhere

### ISSUE THREAD

**Issue by SortaKindaSnek** [CleverRaven/Cataclysm-DDA#29549]
Randomly in towns, a tile on ground level will be a ledge, falling down it places you inside a solid rock wall, and makes you unable to do anything.

I have no idea how to reproduce it, save for looking for them in generated worlds. I have seen two within ~25 worlds, always outside double sized houses.

https://cdn.discordapp.com/attachments/532293215244517379/566772592359505921/Screenshot_20190413-193959.png <A screenshot of the more recent one.

**Versions and configuration(please complete the following information):**
 - OS: Android
 - Game Version: 0.6-26323-gc6b7910c3e
 - Graphics version: Tiles
 - Mods loaded: All available in the base game.

**Comment by snipercup** [comment]
Cannot reproduce the issue on windows, default settings. I noticed your issue came up in a bend at a small town so I set the city size low and found many bends with houses nearby but this is what I see:
![image](https://user-images.githubusercontent.com/50166150/66262624-f45ff580-e7e4-11e9-9e2f-ac6cfec418d2.png)

![image](https://user-images.githubusercontent.com/50166150/66262625-f75ae600-e7e4-11e9-9f1e-69a8c6d45fb4.png)


Grass is in places where one would not expect it. My guess is that something goes wrong in the mapgen when a bend and a house come together.

If someone on android could test again with low city size that'd be great.
